src/Mod/PartDesign/App/FeatureLoft.cpp

```diff
@@ -102,80 +133,109 @@ App::DocumentObjectExecReturn *Loft::execute(void)
-        auto multisections = Sections.getValues();
+        auto multisections = Sections.getSubListValues();
         if (multisections.empty())
             return new App::DocumentObjectExecReturn("Loft: At least one section is needed");
 
-        std::vector<std::vector<TopoDS_Wire>> wiresections;
-        for (TopoDS_Wire& wire : wires)
-            wiresections.emplace_back(1, wire);
-
-        for (auto obj : multisections) {
-            if (!obj->isDerivedFrom(Part::Feature::getClassTypeId()))
-                return  new App::DocumentObjectExecReturn("Loft: All sections need to be part features");
-
-            // if the section is an object's face then take just the face
-            TopoDS_Shape shape;
-            if (obj->isDerivedFrom(Part::Part2DObject::getClassTypeId()))
-                shape = static_cast<Part::Part2DObject*>(obj)->Shape.getValue();
-            else {
-                auto subValues = Sections.getSubValues(obj);
-                if (subValues.empty())
-                    throw Base::ValueError("Loft: No valid subelement linked in Part::Feature");
+        TopoDS_Shape profileShape = getSectionShape(Profile.getValue(),
+                                                    Profile.getSubValues());
+        if (profileShape.IsNull())
+            return new App::DocumentObjectExecReturn("Loft: Could not obtain profile shape");
 
-                shape = static_cast<Part::Feature*>(obj)->Shape.getShape().getSubShape(subValues[0].c_str());
-            }
+        std::vector<std::vector<TopoDS_Shape>> wiresections;
 
+        size_t numWires = addWiresToWireSections(profileShape, wiresections);
+        if (numWires == 0) {
+            // profileShape had no wires so only other valid option is point section
             TopExp_Explorer ex;
-            size_t i=0;
-            for (ex.Init(shape, TopAbs_WIRE); ex.More(); ex.Next(), ++i) {
-                if (i>=wiresections.size())
-                    return new App::DocumentObjectExecReturn("Loft: Sections need to have the same amount of inner wires as the base section");
-                wiresections[i].push_back(TopoDS::Wire(ex.Current()));
+            size_t i = 0;
+            for (ex.Init(profileShape, TopAbs_VERTEX); ex.More(); ex.Next(), ++i) {
+                profilePoint = ex.Current();
             }
-            if (i<wiresections.size())
-                    return new App::DocumentObjectExecReturn("Loft: Sections need to have the same amount of inner wires as the base section");
+            if (i > 1)
+                return new App::DocumentObjectExecReturn("Loft: Only one isolated point is needed if using a sketch with isolated points for section");
+        }
 
+        bool isLastSectionVertex = false;
+
+        for (auto &subSet : multisections) {
+            if (!subSet.first->isDerivedFrom(Part::Feature::getClassTypeId()))
+                return new App::DocumentObjectExecReturn("Loft: All sections need to be part features");

```

Comment:
I would omit the prefix "Loft" since these messages only appear while editing/creating lofts.

Reply:
Are they not shown if you recompute and it fails? It may not be immediately clear from were the error comes from in that case, further more, to me it's better to be always clear what is sending output to the report view, you can then look back on previous messages and know were they came from even if you are working on something else now. In truth I would like it if the output mentioned wich specific feature in three prompted it but that's another discussion.

Reply:
OK, then keep it.

Reply:
I'd agree with @adrianinsaval here. There doesn't appear to be much negative to keeping the "Loft: " in.